### PR TITLE
chore: Switch to Foundry (SC-4852)

### PR DESCRIPTION
**Pull request by JGcarv**
# Description

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/test/ERC20.t.sol [hunk 1]
```diff
@@ -93,7 +93,7 @@ contract ERC20Test is DSTest {
         }
     }
 
-    function proveFail_transfer_insufficientBalance(address to, uint256 mintAmount, uint256 sendAmount) public {
+    function testFail_transfer_insufficientBalance(address to, uint256 mintAmount, uint256 sendAmount) public {
```
Comment (lucas-manuel) [inline contracts/test/ERC20.t.sol]: lets update these to use vm.expectRevert

Comment (lucas-manuel) [reply contracts/test/ERC20.t.sol]: Also lets just fuzz one amount and use the edge case

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Pull library deps
+        run: forge update
+
+      - name: Run forge tests
+        run: ./test.sh -r 100000
```
Comment (lucas-manuel) [inline .github/workflows/pull-request.yml]: Need to use foundry.toml and profiles - refer to rdt

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Small changes
